tests/auth_tests/models/with_integer_username.py

```diff
@@ -0,0 +1,26 @@
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+from django.db import models
+
+
+class CustomUserWithIntegerUsernameManager(BaseUserManager):
+    def create_user(self, username, password):
+        user = self.model(username=username)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def get_by_natural_key(self, username):
+        return self.get(username=username)
+
+
+class CustomUserWithIntegerUsername(AbstractBaseUser):
+    objects = CustomUserWithIntegerUsernameManager()
+
+    username = models.IntegerField()
+    password = models.CharField(max_length=255)
+
+    USERNAME_FIELD = 'username'
+    REQUIRED_FIELDS = ['username', 'password']
+
+    def __str__(self):
```

Comment:
I think `__str__` can be omitted here. If it needs to stay, `CustomUserWithIntegerUsername` should be wrapped by `@python_2_unicode_compatible`.